.adr/0002-signature-scheme.md

```diff
@@ -0,0 +1,37 @@
+## 0002 -- State Signing Algorithm
+
+## Status
+
+Accepted
+
+## Context
+
+We require an algorithm for digitally signing a state. A state is typed data, so the algorithm must encode it, possibly hash it, and then sign using e.g. ECDSA. The algorithm should protect user funds in the light of possibly malicious on-chain code.
+
+## Decision
+
+Our state signing scheme involves encoding a data structure into a string of bytes, hashing those using [`keccak256`](https://en.wikipedia.org/w/index.php?title=Keccak-256&redirect=no) and signing the hash using an ephemeral [ECDSA](https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm) key.
+
+This key should not be used in any other protocol.
+
+Nitro clients should verify the decoding from the contract where funds are locked.
+
+## Consequences
+
+Signing a hash is important because it protects against leakage of the private key -- since the hash is unpredicatble it cannot be crafted maliciously by an adversary (even if the preimage has been prepared by that adversary).
```

Comment:
```suggestion
Signing a hash is important because it protects against leakage of the private key -- since the hash is unpredictable it cannot be crafted maliciously by an adversary (even if the preimage has been prepared by that adversary).
```